src/main/java/battlestates/BattleStateManager.java

```diff
@@ -0,0 +1,107 @@
+     *  Attributes:
+     *  user: Player object representing the user
+     *  foe: EnemyFacade object representing who the user is battling
+     */
+    private Player user;
+    private EnemyFacade foe;
+    private boolean awaitingInput = false;
+    private final OutputHandler output = Output.getScreen();
+
+    public BattleStateManager(Player user) {
+        this.user = user;
+        initialize();
+    }
+
+    /**
+     * Returns the next state based on the current state. If either EnemyFacade or user are dead, ends battle and
+     * provides win or lose state accordingly.
+     * @param input string input to dictate the next state, currently unused
+     * @return the state that comes next in the battle. someitmes when the user enters the wrong number this class would be used to fix the difference between the se
+     */
+    @Override
+    protected State nextState(String input) {
+        boolean userNext = user.getSpeed() >= foe.getSpeed();
+        State chosenState;
+
+        // Win condition
+        if (!foe.checkAlive()) {
+            chosenState = new WinBattleState(user, foe);
+            this.isDone = true;
+        }
+        // Lose condition
+        else if (user.getCurrHealth() <= 0) {
+            chosenState = new LoseBattleState(user, foe);
+            this.isDone = true;
+        }
+        // Normal battle turn ordering, characters can get multiple turns in a row if high enough speed.
+        else {
+            if (userNext) {
+                output.generateText("You outsped the enemy. What would you like to do?");
+                chosenState = new UserTurnState(user, foe);
+            } else {
+                output.generateText(foe.getName() + " outsped you!");
+                chosenState = new EnemyTurnState(user, foe, input);
+            }
+            awaitingInput = chosenState.awaitInput();
+        }
+        return chosenState;
+    }
+    @Override
+    public void preInput() {
+        currState.preInput();
+        awaitingInput = currState.awaitInput(); // Making sure await is updated
+        boolean currPreInput = currState.isDone();
+        if (currPreInput) {
+            this.currState = this.nextState("");
+            if (this.currState == null) {
+                this.isDone = true;
+            }
+        }
+    }
+    @Override
+    public void postInput(String input) {
+        currState.postInput(input);
+        awaitingInput = currState.awaitInput(); // Making sure await is updated
+        boolean currPostInput = currState.isDone();
+        if (currPostInput) {
+            this.currState = this.nextState(input);
+            if (this.currState == null) {
+                this.isDone = true;
+            }
+        }
+    }
+
+    @Override
+    public boolean awaitInput() {
+        return awaitingInput;
+    }
+    @Override
+    public void initialize() {
+        EnemyFactory enemyFactory = new EnemyFactory();
```

Comment:
In the final manager, you will get the name of the enemy from the EventDatabase so you would create the enemy depending on that name.